dsl/jpql/src/main/kotlin/com/linecorp/kotlinjdsl/dsl/jpql/Jpql.kt

```diff
@@ -632,6 +632,22 @@ open class Jpql : JpqlDsl {
         return Expressions.sqrt(value.toExpression())
     }
 
+    /**
+     * Creates an expression that the number of elements of the collection.
+     */
+    @SinceJdsl("3.4.0")
+    fun <T : Any, V> size(expr: KProperty1<T, @Exact V>): Expression<Int> {
+        return Expressions.size(Paths.path(expr))
+    }
+
+    /**
+     * Creates an expression that the number of elements of the collection.
+     */
+    @SinceJdsl("3.4.0")
+    fun <T : Any> size(path: Pathable<T>): Expression<Int> {
+        return Expressions.size(path.toPath())
+    }
```

Comment:
Using a non-collection type as an argument value to the size(..) function results in a runtime error.
For developer experience, why not add a collection type constraint to the generic type parameter so that a compilation error can occur instead of a runtime error?